src/osd/OSD.cc

```diff
@@ -8727,9 +8727,9 @@ void OSD::ShardedOpWQ::_process(uint32_t thread_index, heartbeat_handle_d *hb )
   assert(NULL != sdata);
   sdata->sdata_op_ordering_lock.Lock();
   if (sdata->pqueue->empty()) {
-    sdata->sdata_op_ordering_lock.Unlock();
     osd->cct->get_heartbeat_map()->reset_timeout(hb, 4, 0);
     sdata->sdata_lock.Lock();
```

Comment:
these two locks have different functions, and if sdata->pqueue is empty, the earlier the better to release sdata_op_oerdering_lock, Don't know why just release sdata_op_oerdering_lock after hold stada_lock will solve "missing wakeup signal"